proof.go

```diff
@@ -92,18 +92,18 @@ func (pin proofInnerNode) Hash(childHash []byte) []byte {
 
 //----------------------------------------
 
-type proofLeafNode struct {
+type ProofLeafNode struct {
```

Comment:
Now that we're exporting this, it should be documented as well.

Reply:
I agree, and I started writing godoc for it, before I realized I didn't know enough about how these proofs work to write more than a cursory godoc. ("ProofLeafNode provides a merkle proof for a leaf node" is not terribly helpful.)

None of our proof types are documented anywhere at the moment, and this lack of clarity on my part was the motivation for opening https://github.com/tendermint/iavl/issues/204. Godoc for proofs can either happen along with that issue or right after it.

Reply:
Ah, right. Sure, let's document it then.